Source/UI/NewUI/PaymentForm/View/CardNumber/CardNumberView.swift

```diff
@@ -52,13 +69,31 @@ final class CardNumberView: UIView, CardNumberViewProtocol {
   private func updateIcon() {
     cardNumberInputView.update(image: schemeIcon.image, animated: true)
   }
+
+  private func updateError() {
+    guard var style = style else { return }
+    style.error?.isHidden = cardNumberError == nil
+
+    update(style: style)
+  }
 }
 
 extension CardNumberView: TextFieldViewDelegate {
   func textFieldShouldBeginEditing(textField: UITextField) { }
   func textFieldShouldChangeCharactersIn(textField: UITextField, replacementString string: String) { }
   func textFieldShouldReturn() -> Bool { true }
-  func textFieldShouldEndEditing(textField: UITextField, replacementString: String) -> Bool { true }
+
+  func textFieldShouldEndEditing(textField: UITextField, replacementString: String) -> Bool {
+    switch viewModel.validate(cardNumber: replacementString) {
+    case .success(let schemeIcon):
+      self.schemeIcon = schemeIcon
+      self.cardNumberError = nil
+    case .failure(let cardNumberError):
+      self.cardNumberError = cardNumberError
```

Comment:
Together with the other suggestion, I wouldn't hold the error, just call the method I suggested, on both paths.

❗ if I start inputting a VISA number and start deleting, you will not remove the scheme icon until I enter a different scheme. So I can have an input of 42, with a Mastercard logo, which may appear confusing.